File: 백준/Gold/2879. 코딩은 예쁘게/코딩은 예쁘게.py
import sys

N = int(sys.stdin.readline())
current_tab = list(map(int, sys.stdin.readline().split()))
correct_tab = list(map(int, sys.stdin.readline().split()))

def make_correct_indent(n, current_tab, correct_tab):
    diff = []
    for i in range(n):
        diff.append(correct_tab[i] - current_tab[i])

    temp = [0] * n
    temp[0] = abs(diff[0])

    i = 1
    while i < n:
        if diff[i] * diff[i - 1] > 0:
            temp[i] = temp[i - 1] + max(0, abs(diff[i]) - abs(diff[i - 1]))
        else:
            temp[i] = temp[i - 1] + abs(diff[i])
        i += 1
    return temp[-1]


print(make_correct_indent(N, current_tab, correct_tab))

File: 백준/Gold/2879. 코딩은 예쁘게/코딩은 예쁘게.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int[] currentGroup = new int[N];
        int[] correctGroup = new int[N];
        int[] diff = new int[N];

        String[] tempArr = br.readLine().split(" ");
        for (int i = 0; i < N; i++){
            currentGroup[i] = Integer.parseInt(tempArr[i]);
        }
        tempArr = br.readLine().split(" ");
        for (int i = 0; i < N; i++){
            correctGroup[i] = Integer.parseInt(tempArr[i]);
            diff[i] = correctGroup[i] - currentGroup[i];
        }

        int[] count = new int[N];
        count[0] = Math.abs(diff[0]);

        for (int i = 1; i < N; i++){
            if (diff[i] * diff[i - 1] > 0){
                count[i] = count[i - 1] + Math.max(0, Math.abs(diff[i]) - Math.abs(diff[i-1]));
            } else {
                count[i] = count[i - 1] + Math.abs(diff[i]);
            }
        }


        System.out.println(count[N-1]);
    }
}
